sequencer/src/block.rs

```diff
@@ -752,29 +723,30 @@ mod test_headers {
             assert!(self.expected_l1_head >= self.parent_l1_head);
             assert!(self.expected_l1_finalized >= self.parent_l1_finalized);
 
-            let genesis = Header::genesis().0;
+            let state = NodeState {};
+            let (genesis, _, metadata) = Header::genesis(&state);
 
             let mut parent = genesis.clone();
             parent.timestamp = self.parent_timestamp;
             parent.l1_head = self.parent_l1_head;
             parent.l1_finalized = self.parent_l1_finalized;
             let block_merkle_tree =
-                BlockMerkleTree::from_elems(32, Vec::<Commitment<Header>>::new()).unwrap();
+                BlockMerkleTree::from_elems(Some(32), Vec::<Commitment<Header>>::new()).unwrap();
             let block_merkle_tree_root = block_merkle_tree.commitment();
 
             let FeeReceipt { recipient, amount } = FeeReceipt::default();
             let fee_merkle_tree =
                 FeeMerkleTree::from_kv_set(20, Vec::from([(recipient, amount)])).unwrap();
             let fee_merkle_tree_root = fee_merkle_tree.commitment();
 
-            let validated_state = ValidatedState {
+            let _validated_state = ValidatedState {
                 block_merkle_tree,
                 fee_merkle_tree,
             };
```

Comment:
If this is not used anywhere, should we delete it?

Reply:
fixed in c8440bfff33f85094ae1635a4c0761a4be741527